encode_utils/connection.py

```diff
@@ -103,6 +103,9 @@ def __init__(self, dcc_mode=None, dry_run=False, submission=False, no_log_file=F
         for name, host in eu.DCC_MODES.items():
             self.dcc_modes.add_mode(host["url"], mode_name=name)
         self._dcc_mode = dcc_mode
+        if self._dcc_mode is None:
+            dcc_mode = self._get_dcc_mode_from_env()
+            self._dcc_mode = dcc_mode
```

Comment:
I'd prefer not to move this code into the `__init__`, as it has side effects (hence why it's currently run lazily in the `dcc_mode` `@property`. I don't think it's strictly necessary to change the way this works for this PR?

Reply:
Hay @paul-sud. Thanks for the comment.
The problem comes from  `_get_logfile_name` being called before `_dcc_mode` is set. That's why I moved that code.
I could either call the property once before that functions is called, but it does not seem very neat to me... Any suggestion?

Reply:
look [here](https://github.com/StanfordBioinformatics/encode_utils/blob/master/encode_utils/connection.py#L205)

Reply:
Another option would be to add a `name` property to the `DccMode` object

Reply:
Ah I see now. Apart from a large refactor of the logging stuff, a quick and dirty way to do it would be to make an additional call to `_get_dcc_mode_from_env` within `_get_logfile_name`

Reply:
OK. And what about using the `host` property of `DccMode`?

Reply:
If you do that you still end up with the same problem of side effects during initialization unfortunately

Reply:
I see. So I just used `_get_dcc_mode_from_env` as you suggested. I also squashed all changes in one commit for convenience.
Just out of curiosity, what could those side effects be like?

Reply:
In this case, reading from the environment and writing to log files.